ScoreSaber/UI/Elements/Profile/ProfileDetailView.cs

```diff
@@ -172,6 +295,91 @@ internal async Task ShowProfile(string playerId) {
             SetProfileBadges(list.ToArray());
             SetCrowns(playerId);
             SetLoadingState(false);
+            if(_beatmapDownloader == null) {
+                Plugin.Log.Error("BeatmapDownloader is null, install a mod that injects IScoreSaberBeatmapDownloader");
+                return;
+            }
+            try {
+                _richPresence = await _playerService.GetRichPresence(playerId);
+                if(_richPresence.state.currentMap != null) {
+                    mapName = _richPresence.state.currentMap.Name;
+                    mapArtist = $"<color=grey><size=80%>{_richPresence.state.currentMap.Artist}</size></color>";
+                    MapMapperName = $"[<color=#59cf59><size=80%>{_richPresence.state.currentMap.AuthorName}</size></color>]";
+                    DateTimeOffset parsedTime = DateTimeOffset.Parse(_richPresence.state.currentMap.Timestamp);
+                    DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                    TimeSpan difference = now - parsedTime;
+                    mapTime = difference.ToNaturalTime(2, false) + " ago";
+                    mapPresence.gameObject.SetActive(true);
+                    mapPresenceTitle.gameObject.SetActive(true);
+                    mapImagePresence.SetImageAsync($"https://cdn.scoresaber.com/covers/{_richPresence.state.currentMap.Hash}.png").RunTask();
+                }
+                Plugin.Log.Notice(_richPresence.state.Scene.ToString());
+                if (_richPresence != null) {
+                    SetRichStatus(_richPresence.state.Scene);
+                }
+            } catch(HttpErrorException ex) {
+                Plugin.Log.Error(ex.Message);
+            }
+        }
+
+        [UIAction("map-presence-play")]
+        private void MapPresenceClicked() {
+            _scoreSaberLeaderboardViewController.CloseModals();
+            _parserParams.EmitEvent("close-modals");
+            DownloadOrOpenMap(_richPresence.state.currentMap.Hash, _richPresence.state.currentMap.Difficulty).RunTask();
+        }
+
+        private async Task DownloadOrOpenMap(string hash, int diff) {
+            if (_beatmapLevelsModel.GetBeatmapLevel("custom_level_" + hash) != null) {
```

Comment:
this should not work if the person is playing an OST/DLC map.

Reply:
fixed